lib/services/appmanager-service.ts

```diff
@@ -50,14 +52,52 @@ class AppManagerService implements IAppManagerService {
 			let projectName = this.$project.projectData.ProjectName;
 			let solutionPath = buildResult[0].solutionPath;
 			let projectPath = solutionPath.substr(solutionPath.indexOf("/") + 1);
-			this.$server.tam.uploadApplication(projectName, projectName, projectPath).wait();
 
-			this.$logger.info("Successfully uploaded package.");
+			let publishSettings: Server.PublishSettings = {
+				IsPublished: options.publish,
+				NotifyByPush: options.sendPush,
+				NotifyByEmail: options.sendEmail,
+				Groups: []
+			};
+
+			if (options.group.length) {
+				publishSettings.Groups = this.findGroups(options.group).wait();
+			}
 
+			this.$server.tam.uploadApplication(projectName, projectName, projectPath, publishSettings).wait();
+			this.$logger.info("Successfully uploaded package.");
 			this.openAppManagerStore();
 		}).future<void>()();
 	}
 
+	public getGroups(): IFuture<void> {
+		return ((): any  => {
+			this.$project.ensureProject();
```

Comment:
no, no, please don't 
Requiring project is fine, this doesn't guarantee that you should execute the command inside project. There was a bug in our code, which I've fixed some hours ago, which was causing the failure when you are not inside project. If you remove the line, I believe you'll be able to execute `appbuilder appmanager groups` outside of a project. 
(You should keep ensureLoggedIn call)